AutoCollection/ClientRequests.ts

```diff
@@ -73,8 +73,8 @@ class AutoCollectClientRequests {
      * Tracks an outgoing request. Because it may set headers this method must be called before
      * writing content to or ending the request.
      */
-    public static trackRequest(client: Client, requestOptions: any, request: http.ClientRequest,
-            properties?: { [key: string]: string }) {
+    public static trackRequest(client: Client, requestOptions: http.RequestOptions, request: http.ClientRequest,
```

Comment:
Consider adding a host property to the `ClientRequestParser` (and move the call to it up before line 88) so that you can re-use its request options parsing instead of duplicating it. Or something like that.